build.ps1

```diff
@@ -0,0 +1,290 @@
+    return $args
+}
+
+function ExecuteBuild() {
+    $pyInstallResults = ''
+    $installArgs = [System.Collections.ArrayList]@()
+
+    if ($pyinstaller) { $installArgs = BuildPyInstaller }
+    else { $installArgs = BuildNuitka }
+
+    if ($debug) { echo "InstallArgs: $installArgs" }
+
+    # -- Turn formatter off so as not to goof up the version number text
+    # @formatter:off
+    $doInstall = {
+        param([System.Collections.ArrayList]$arguments)
+        try {
+            python -m pip install --upgrade pip
+            python -m pip install -r requirements.txt --extra-index-url https://pypi.bartbroe.re
+            python -m pip install --extra-index-url https://pypi.bartbroe.re openimageio==2.2.18.0 scons
+
+            if ($debug) { echo "arguments: $arguments" }
+            $argString = $($arguments | % { "$_" })
+
+            if ($debug) { echo "argString: $argString" }
+            $cmd = "${argString}"
+
+            if ($debug) { echo "cmd: ${cmd}" }
+            iex ${cmd}
+
+            if ($pyinstaller) { $dest = "dist/GimelStudio/" }
+            else {
+                $dest = "build/main.dist/";
+                mv -path "build/main.dist/main.exe" -destination "build/main.dist/GimelStudio.exe"
+            }
+            cp -recurse -force -path $nodes -destination $dest
+
+        }
+        catch { $_; $pyInstallResults = $_; "Error installing OpenImageIO. GimelStudio will not work without it. Exiting..."; exit 1; }
+    } # @formatter:on
+
+    try { & $doInstall $installArgs }
+    catch { $_; "Error installing OpenImageIO. GimelStudio will not work without it. Exiting..."; exit 1; }
+
+    # -- Check for errors on installation of components ---
+    if (!([string]::IsNullOrEmpty($pyInstallResults)) -and ($pyInstallResults -match "OpenImageIO")) {
+        "Error installing OpenImageIO. GimelStudio will not work without it. Exiting..."; exit 1;
+    }
+}
+
+# --| Declarations ----------------------------------------
+# --|------------------------------------------------------
+$originalPath = $env:PATH
+try {
+    $tmpPath = [System.Collections.ArrayList]@()
+    # @formatter:off
+    $tmpPath = @(
+        'C:\Windows\System32\Wbem;',
+        'C:\Program Files\PowerShell\7;',
+        'C:\Program Files\PowerShell\7-preview;',
+        'C:\Program Files\Git\cmd\;',
+        'C:\Windows\System32\;',
+        'C:\Windows;',
+        'C:\Windows\System32\WindowsPowerShell\v1.0'
+    )
+    $env:PATH = $($($tmpPath | % { "$_".Trim() }) -join (''))
+    # @formatter:on
+
+    [switch]$debug = $doDebug
+
+    # --| Main Setup Start ------------------------------------
+    # --|------------------------------------------------------
+    if ($interactive) {
+        echo "GimelStudio build process initiation for missing dependencies"
+        echo "=============================================================\n\n"
+        echo 'This script will download and install the required dependencies for the build process'
```

Comment:
This is a small but perhaps important detail. Could you just add a list of the dependencies here for the user to see? That way its transparent to the user what will be installed ahead of time.

Reply:
Sure thing, can do. All deps are technically temporary, installed into .conda and .tools folder, which could simply be deleted after build if desired. No permanent changes are made to the $PATH, it is all setup within the shell session.

Reply:
Sure thing, can do. All deps are technically temporary, installed into .conda and .tools folder, which could simply be deleted after build if desired. No permanent changes are made to the $PATH, it is all setup within the shell session.